File: backend/api/src/main/scala/com/github/ai/split/api/CurrencyDto.scala
package com.github.ai.split.api

import zio.json.{DeriveJsonEncoder, JsonEncoder}

case class CurrencyDto(
  isoCode: String,
  name: String,
  symbol: String
)

object CurrencyDto {
  implicit val encoder: JsonEncoder[CurrencyDto] = DeriveJsonEncoder.gen[CurrencyDto] 
}

File: backend/api/src/main/java/com/github/ai/split/api/CurrencyDto.java
package com.github.ai.split.api;

public class CurrencyDto {

    public String isoCode = "";
    public String name = "";
    public String symbol = "";

    public CurrencyDto() {}

    public CurrencyDto(String isoCode, String name, String symbol) {
        this.isoCode = isoCode;
        this.name = name;
        this.symbol = symbol;
    }
}
